src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceDescription.cs

```diff
@@ -77,11 +80,24 @@ public string ConfigurationName
 
         public Type ServiceType { get; set; }
 
-        internal static void AddBehaviors<TService>(ServiceDescription serviceDescription) where TService : class
+        internal static void AddBehaviors<TService>(ServiceDescription serviceDescription, IEnumerable<IServiceBehavior> injectedBehaviors = null) where TService : class
         {
             TypeLoader<TService>.ApplyServiceInheritance<IServiceBehavior, KeyedByTypeCollection<IServiceBehavior>>(
                 serviceDescription.Behaviors, GetIServiceBehaviorAttributes);
 
+            if (injectedBehaviors != null)
```

Comment:
how this is different from ConfigureServiceHostBase ?

Reply:
ConfigureServiceHostBase allows you to provide a delegate which is passed the ServiceHostBase instance for you to modify. This is the mechanism of allowing you to add an implementation of IServiceBehavior to DI and have that automatically added to the description. The problem was that the DI behaviors were being added too late which meant if you tried to add a ServiceBehaviorAttribute to DI, it would fail because the automatic add if missing logic had already ran.